### PR TITLE
Update Ehcache to 3.10.0

### PR DESCRIPTION
**Pull request by reta**
@coheigea ~seems like https://github.com/apache/ws-wss4j/commit/e4a33efcb2b474a1da2b2c08f815b2718e111823 fixes repositories issue~ I had to exclude the jaxb dependencies from Ehcache

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -36,11 +36,6 @@ jobs:
         # a pull request then we can checkout the head.
         fetch-depth: 2
 
-    # If this run was triggered by a pull request event, then checkout
-    # the head of the pull request instead of the merge commit.
-    - run: git checkout HEAD^2
```
Comment (reta) [inline .github/workflows/codeql-analysis.yml:41]:
```
Run github/codeql-action/init@v1
Warning: 1 issue was detected with this workflow: git checkout HEAD^2 is no longer necessary. Please remove this step as Code Scanning recommends analyzing the merge commit for best results.
```

**Comment by coheigea** [comment]
Thanks @reta . Should it have the jakarta classifier? I see:

[INFO] +- org.ehcache:ehcache:jar:3.10.0:compile (optional)
[INFO] |  \- javax.cache:cache-api:jar:1.1.0:compile (optional)

**Comment by reta** [comment]
Oh yeah, this is the Jakarta-compatible version, updated, thanks @coheigea !

**Comment by jimma** [comment]
Thanks the change, @reta  and @coheigea !  This will fix the test failure in CXF without the ehcache exclusion for wss4j dependency:
```
Caused by: java.lang.NoClassDefFoundError: javax/xml/bind/ValidationEventHandler
	at org.ehcache.xml.XmlConfiguration.<init>(XmlConfiguration.java:114)
	at org.apache.cxf.ws.security.tokenstore.EHCacheTokenStore.<init>(EHCacheTokenStore.java:69)
	at org.apache.cxf.sts.cache.DefaultInMemoryTokenStore.<init>(DefaultInMemoryTokenStore.java:36)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:206)
	... 59 more

```
 I saw this PR until I submitted my PR to do the same thing. I already closed mine.   There is a jira for this upgrade: https://issues.apache.org/jira/browse/WSS-696.  Feel free to close it if it's not needed.

**Review by coheigea — APPROVED** [review]

**Pull request merged**